src/sql/workbench/contrib/notebook/browser/notebookFindWidget.ts

```diff
@@ -0,0 +1,574 @@
+
+		MAX_MATCHES_COUNT_WIDTH = Math.max(MAX_MATCHES_COUNT_WIDTH, this._matchesCount.clientWidth);
+	}
+
+	// ----- actions
+
+	private _updateButtons(): void {
+		this._findInput.setEnabled(this._isVisible);
+		this._closeBtn.setEnabled(this._isVisible);
+
+		let findInputIsNonEmpty = (this._state.searchString.length > 0);
+		this._prevBtn.setEnabled(this._isVisible && findInputIsNonEmpty);
+		this._nextBtn.setEnabled(this._isVisible && findInputIsNonEmpty);
+	}
+
+	private _reveal(animate: boolean): void {
+		if (!this._isVisible) {
+			this._isVisible = true;
+
+			this._updateButtons();
+
+			setTimeout(() => {
+				dom.addClass(this._domNode, 'visible');
+				this._domNode.setAttribute('aria-hidden', 'false');
+				if (!animate) {
+					dom.addClass(this._domNode, 'noanimation');
+					setTimeout(() => {
+						dom.removeClass(this._domNode, 'noanimation');
+					}, 200);
+				}
+			}, 0);
+			this._notebookController.layoutOverlayWidget(this);
+		}
+	}
+
+	private _hide(focusTheEditor: boolean): void {
+		if (this._isVisible) {
+			this._isVisible = false;
+
+			this._updateButtons();
+
+			dom.removeClass(this._domNode, 'visible');
+			this._domNode.setAttribute('aria-hidden', 'true');
+			if (focusTheEditor) {
+				this._notebookController.focus();
+			}
+			this._notebookController.layoutOverlayWidget(this);
+		}
+	}
+
+	private _applyTheme(theme: ITheme) {
+		let inputStyles: IFindInputStyles = {
+			inputActiveOptionBorder: theme.getColor(colors.inputActiveOptionBorder),
+			inputBackground: theme.getColor(colors.inputBackground),
+			inputForeground: theme.getColor(colors.inputForeground),
+			inputBorder: theme.getColor(colors.inputBorder),
+			inputValidationInfoBackground: theme.getColor(colors.inputValidationInfoBackground),
+			inputValidationInfoBorder: theme.getColor(colors.inputValidationInfoBorder),
+			inputValidationWarningBackground: theme.getColor(colors.inputValidationWarningBackground),
+			inputValidationWarningBorder: theme.getColor(colors.inputValidationWarningBorder),
+			inputValidationErrorBackground: theme.getColor(colors.inputValidationErrorBackground),
+			inputValidationErrorBorder: theme.getColor(colors.inputValidationErrorBorder)
+		};
+		this._findInput.style(inputStyles);
+	}
+
+	// ----- Public
+
+	public focusFindInput(): void {
+		this._findInput.focus();
+	}
+
+	public highlightFindOptions(): void {
+		this._findInput.highlightFindOptions();
+	}
+
+	private _onFindInputMouseDown(e: IMouseEvent): void {
+		// on linux, middle key does pasting.
+		if (e.middleButton) {
+			e.stopPropagation();
+		}
+	}
+
+	private _onFindInputKeyDown(e: IKeyboardEvent): void {
+
+		if (e.equals(KeyCode.Enter)) {
+			this._notebookController.getAction(ACTION_IDS.FIND_NEXT).run().then(null, onUnexpectedError);
+			e.preventDefault();
+			return;
+		}
+
+		if (e.equals(KeyMod.Shift | KeyCode.Enter)) {
+			this._notebookController.getAction(ACTION_IDS.FIND_NEXT).run().then(null, onUnexpectedError);
```

Comment:
Shouldn't this be FIND_PREVIOUS?

Reply:
I believe it should but Lucy copied this code over from profilerFindWidget and that has FIND_NEXT for both the keyboard events.

Reply:
I'd consider that a bug - I'll go and fix up the profiler but you should also fix it up here to do the previous if shift + enter is pressed.

Reply:
already did, fixing and cleaning up of couple of other things. Will push all of them in a while.